lib/datasource/docker/index.ts

```diff
@@ -535,3 +543,34 @@ export async function getPkgReleases({
   }
   return ret;
 }
+
+function getECRAuthToken(region: string, opts: hostRules.HostRule) {
+  const config = { region, accessKeyId: undefined, secretAccessKey: undefined };
+  if (opts.username && opts.password) {
+    config.accessKeyId = opts.username;
+    config.secretAccessKey = opts.password;
+  }
+  const ecr = new AWS.ECR(config);
+  return new Promise<string>((resolve, reject) => {
+    ecr.getAuthorizationToken({}, (err, data) => {
+      if (err) {
+        logger.warn({ err });
+        resolve(null);
+      }
+      const authorizationToken =
+        data &&
+        data.authorizationData &&
+        data.authorizationData[0] &&
+        data.authorizationData[0].authorizationToken;
+      if (authorizationToken) {
+        resolve(authorizationToken);
+      } else {
+        logger.error(
+          { data },
```

Comment:
There is the risk of this leaking a token here if somehow the token is in another field in the `data` object

Reply:
I understand the point and have removed it, but the tradeoff though is low value logs which won't aid much when debugging.